src/Adapter/Product/Image/Repository/ProductImageRepository.php

```diff
@@ -364,4 +392,76 @@ public function delete(Image $image): void
     {
         $this->deleteObjectModel($image, CannotDeleteProductImageException::class);
     }
+
+    /**
+     * @param ProductId $productId
+     * @param LanguageId $languageId
+     *
+     * @return ProductPreview
+     *
+     * @throws CoreException
+     */
+    public function getProductCoverUrl(ProductId $productId, LanguageId $languageId): ProductPreview
```

Comment:
I think I wasn't clear enough in my previous comment The purpose was not to just rename the function but to split it in two:
- one method returns the image URL **only** so no need to fetch the whole Product to just get its name (`getProductCoverUrl`)
- one method returns a `ProductPreview` so it returns both the image and the name for preview purposes (`getProductPreview`)

Reply:
The same principle should be applied for combination as well And ideally if we could fetch the product name without fetching the whole product in `ProductpreviewRepository` it would be perfect then